Sources/BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -295,21 +281,18 @@ public class KeyringStore: ObservableObject, WalletObserverStore {
     }
     Task { @MainActor in // fetch all KeyringInfo for all coin types
       let allAccounts = await keyringService.allAccounts()
-      let allKeyrings = await keyringService.keyrings(for: WalletConstants.supportedCoinTypes())
-      if let defaultKeyring = allKeyrings.first(where: { $0.id == BraveWallet.KeyringId.default }) {
-        self.defaultKeyring = defaultKeyring
-        self.isDefaultKeyringLoaded = true
-        self.isDefaultKeyringCreated = defaultKeyring.isKeyringCreated
-        // fallback case where user completed front-end onboarding, but has no keyring created/accounts.
-        if !defaultKeyring.isKeyringCreated && Preferences.Wallet.isOnboardingCompleted.value {
-          Preferences.Wallet.isOnboardingCompleted.reset()
-        }
-      }
-      self.allKeyrings = allKeyrings
       self.allAccounts = allAccounts.accounts
       if let selectedAccount = allAccounts.selectedAccount {
         self.selectedAccount = selectedAccount
       }
+      self.isWalletCreated = await keyringService.isWalletCreated()
+      // fallback case where user completed front-end onboarding, but has no keyring created/accounts.
```

Comment:
Might be a good idea to add in comment why this might happen actually

Reply:
Thanks, added to comment in [addd87b](https://github.com/brave/brave-ios/pull/8313/commits/addd87b1bdf7af9e31080ad8138b8c104367f08d)